### PR TITLE
gh-81057: Add a CI Check for New Unsupported C Global Variables

### LINKED ISSUE #100237: Update test_check_c_globals to not need distutils & reenable it
`distutils` was removed from 3.12, as per PEP 632.
`test_check_c_globals`, which uses it, was disabled. It should be updated to not need `distutils`.

See #92584

### PR DESCRIPTION
**Pull request by ericsnowcurrently**
This will keep us from adding new unsupported (i.e. non-const) C global variables, which would break interpreter isolation.

FYI, historically it is very uncommon for new global variables to get added.  Furthermore, it is rare for new code to break the c-analyzer.  So the check should almost always pass unnoticed.

Note that I've removed test_check_c_globals.  A test wasn't a great fit conceptually and was *super* slow on debug builds.  A CI check is a better fit.

(I'm going to sit on this PR for a few days in case there are any concerns with the new CI check.  I'll also do a quick double-check with the core team but don't anticipate any real opposition.)

This also resolves gh-100237.

* Issue: gh-81057

### CODE REVIEW

**Review by erlend-aasland — APPROVED** [review]
LGTM (but patchcheck has white-space complaints)

**Comment by ericsnowcurrently** [comment]
Yeah, it never likes the tabs in the TSV strings in Tools/c-analyzer/cpython/_parser.py.

**Comment by ericsnowcurrently** [comment]
I started a thread:  https://discuss.python.org/t/adding-a-ci-check-to-detect-new-c-global-variables/24731.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -111,6 +111,9 @@ jobs:
         run: make smelly
       - name: Check limited ABI symbols
         run: make check-limited-abi
+      - name: Check for Unsupported C Global Variables
+        if: github.event_name == 'pull_request'  # $GITHUB_EVENT_NAME
+        run: python3 Tools/c-analyzer/check-c-globals.py --format summary --traceback
```
Comment (CAM-Gerlach) [inline .github/workflows/build.yml]: For consistency with the other checks, perhaps worth making this a `make` target and calling it from there?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -111,6 +111,9 @@ jobs:
         run: make smelly
       - name: Check limited ABI symbols
         run: make check-limited-abi
+      - name: Check for Unsupported C Global Variables
```
Comment (CAM-Gerlach) [inline .github/workflows/build.yml]:
```suggestion
      - name: Check for unsupported C global variables
```

Minor nit: Use the same sentence case as the other step names

**Review by CAM-Gerlach — COMMENTED** [review]
A couple minor comments; I reviewed the check explanation and didn't find any non-trivial copyediting/proofreading issues.

**Review by kumaraditya303 — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (ericsnowcurrently) [reply .github/workflows/build.yml]: done

**Pull request merged**